src/lib/matrix/test/MatrixAssignmentTest.cpp

```diff
@@ -191,28 +222,28 @@ int main()
 	Matrix3f m10_upper_bound(data_m10_upper_bound);
 	Matrix3f m10_upper_bounded_ref(data_m10_upper_bounded_ref);
 	Matrix3f m10_constrained_ref(data_m10_constrained_ref);
-	TEST(isEqual(max(m10, m10_lower_bound), m10_lower_bounded_ref));
-	TEST(isEqual(max(m10_lower_bound, m10), m10_lower_bounded_ref));
-	TEST(isEqual(min(m10, m10_upper_bound), m10_upper_bounded_ref));
-	TEST(isEqual(min(m10_upper_bound, m9), m10_upper_bounded_ref));
-	TEST(isEqual(constrain(m10, m10_lower_bound, m10_upper_bound), m10_constrained_ref));
+	EXPECT_EQ(max(m10, m10_lower_bound), m10_lower_bounded_ref);
+	EXPECT_EQ(max(m10_lower_bound, m10), m10_lower_bounded_ref);
+	EXPECT_EQ(min(m10, m10_upper_bound), m10_upper_bounded_ref);
+	EXPECT_EQ(min(m10_upper_bound, m9), m10_upper_bounded_ref);
+	EXPECT_EQ(constrain(m10, m10_lower_bound, m10_upper_bound), m10_constrained_ref);
 
 	// min, max, constrain with NAN
-	TEST(isEqualF(matrix::typeFunction::min(5.f, NAN), 5.f));
-	TEST(isEqualF(matrix::typeFunction::min(NAN, 5.f), 5.f));
-	TEST(isEqualF(matrix::typeFunction::min(NAN, NAN), NAN));
-	TEST(isEqualF(matrix::typeFunction::max(5.f, NAN), 5.f));
-	TEST(isEqualF(matrix::typeFunction::max(NAN, 5.f), 5.f));
-	TEST(isEqualF(matrix::typeFunction::max(NAN, NAN), NAN));
-	TEST(isEqualF(matrix::typeFunction::constrain(NAN, 5.f, 6.f), NAN));
-	TEST(isEqualF(matrix::typeFunction::constrain(1.f, 5.f, 4.f), NAN));
-	TEST(isEqualF(matrix::typeFunction::constrain(6.f, NAN, 5.f), 5.f));
-	TEST(isEqualF(matrix::typeFunction::constrain(1.f, 5.f, NAN), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::min(5.f, NAN), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::min(NAN, 5.f), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::min(NAN, NAN), NAN));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::max(5.f, NAN), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::max(NAN, 5.f), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::max(NAN, NAN), NAN));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::constrain(NAN, 5.f, 6.f), NAN));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::constrain(1.f, 5.f, 4.f), NAN));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::constrain(6.f, NAN, 5.f), 5.f));
+	EXPECT_TRUE(isEqualF(matrix::typeFunction::constrain(1.f, 5.f, NAN), 5.f));
```

Comment:
No, because EXPECT_FLOAT_EQ(NAN, NAN) fails and in multiple of these checks we want to verify the results really match NAN. Instead, we could use the right implementation of `isnan()` but `isEqualF()` takes care of this for us.